wire-runtime/src/main/java/com/squareup/wire/Wire.java

```diff
@@ -115,59 +115,6 @@ public Wire(List<Class<?>> extensionClasses) {
   }
 
   /**
-   * Reads a message of type {@code messageClass} from {@code bytes} and returns
-   * it.
-   */
-  public <M extends Message> M parseFrom(byte[] bytes, Class<M> messageClass) throws IOException {
```

Comment:
You might have a byte array from a Parcel, for example. Or sometimes a proto message contains a `bytes[]` field that is an encrypted version of another proto message. You don't always have access to a stream.

I've specifically called out this method (and .toByteArray()) as examples of how simple it is to use Wire. If you have to also know something (even just a little) about Okio streams in order to do basic serialization/deserialization, the examples get much less compelling. It would be great to have comments on the convenience methods that explain exactly how they work and how you can gain efficiency by using lower-level primitives in your application code.